modules/datastore/src/Plugin/DatasetInfo/DatastoreInfo.php

```diff
@@ -0,0 +1,157 @@
+
+  /**
+   * Datastore service.
+   */
+  protected DatastoreService $datastore;
+
+  /**
+   * Resource mapper service.
+   */
+  protected ResourceMapper $resourceMapper;
+
+  /**
+   * Constructs a \Drupal\Component\Plugin\PluginBase object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $pluginId
+   *   The plugin_id for the plugin instance.
+   * @param mixed $pluginDefinition
+   *   The plugin implementation definition.
+   * @param \Drupal\datastore\Service\Info\ImportInfo $importInfo
+   *   Import info datastoer service.
+   * @param \Drupal\datastore\DatastoreService $datastore
+   *   Datastore service.
+   * @param \Drupal\metastore\ResourceMapper $resourceMapper
+   *   Resource mapper service.
+   */
+  public function __construct(
+    array $configuration,
+    $pluginId,
+    $pluginDefinition,
+    ImportInfo $importInfo,
+    DatastoreService $datastore,
+    ResourceMapper $resourceMapper,
+  ) {
+    parent::__construct($configuration, $pluginId, $pluginDefinition);
+    $this->importInfo = $importInfo;
+    $this->datastore = $datastore;
+    $this->resourceMapper = $resourceMapper;
+  }
+
+  /**
+   * Container injection.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The service container.
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $pluginId
+   *   The plugin_id for the plugin instance.
+   * @param mixed $pluginDefinition
+   *   The plugin implementation definition.
+   *
+   * @return static
+   */
+  public static function create(
+    ContainerInterface $container,
+    array $configuration,
+    $pluginId,
+    $pluginDefinition,
+  ) {
+    return new static(
+      $configuration,
+      $pluginId,
+      $pluginDefinition,
+      $container->get('dkan.datastore.import_info'),
+      $container->get('dkan.datastore.service'),
+      $container->get('dkan.metastore.resource_mapper'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addDatasetInfo(array $info): array {
+    foreach ($info as &$revision) {
+      if (!isset($revision['distributions'])) {
+        continue;
+      }
+      foreach ($revision['distributions'] as &$distribution) {
+        $this->addDistributionInfo($distribution);
+      }
+    }
+    return $info;
+  }
+
+  /**
+   * Alter the distribution info.
+   *
+   * @param array $distribution
+   *   The distribution info array from a dataset info array.
+   */
+  protected function addDistributionInfo(array &$distribution): void {
+    $identifier = $distribution['resource_id'];
+    $version = $distribution['resource_version'];
+    $import_info = $this->importInfo->getItem($identifier, $version);
+    $fileMapper = $this->resourceMapper->get($identifier, ResourceLocalizer::LOCAL_FILE_PERSPECTIVE, $version);
```

Comment:
You could do `$this->datastore->getResourceMapper()->get($identifier, ResourceLocalizer::LOCAL_FILE_PERSPECTIVE, $version);`

Reply:
Set in constructor